modin/pandas/base.py

```diff
@@ -2759,8 +2759,10 @@ def transform(self, func, axis=0, *args, **kwargs):
         kwargs["is_transform"] = True
         try:
             result = self.agg(func, axis=axis, *args, **kwargs)
-        except Exception:
-            raise ValueError("Transform function failed")
+        except TypeError:
```

Comment:
reflecting the same [pandas changes](https://github.com/pandas-dev/pandas/pull/40288/files#diff-2257b34410aee27eb14e348b9545fef2e212ff93bd72af02d700ae8df43d97bbL230)